azurerm/internal/services/web/app_service_certificate_binding_resource.go

```diff
@@ -0,0 +1,223 @@
+		},
+
+		Schema: map[string]*schema.Schema{
+
+			"hostname_binding_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.HostnameBindingID,
+			},
+
+			"certificate_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.CertificateID,
+			},
+
+			"ssl_state": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(web.SslStateIPBasedEnabled),
+					string(web.SslStateSniEnabled),
+				}, false),
+			},
+
+			"hostname": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"app_service_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"thumbprint": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceArmAppServiceCertificateBindingCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Web.AppServicesClient
+	certClient := meta.(*clients.Client).Web.CertificatesClient
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	log.Printf("[INFO] preparing arguments for App Service Hostname Binding creation.")
+
+	hostnameBindingID := d.Get("hostname_binding_id").(string)
+	certificateID := d.Get("certificate_id").(string)
+
+	id, err := parse.NewCertificateBindingId(hostnameBindingID, certificateID)
+	if err != nil {
+		return fmt.Errorf("could not parse ID: %+v", err)
+	}
+
+	certDetails, err := certClient.Get(ctx, id.CertificateId.ResourceGroup, id.CertificateId.Name)
+	if err != nil {
+		if utils.ResponseWasNotFound(certDetails.Response) {
+			return fmt.Errorf("retrieving App Service Certificate %q (Resource Group %q), not found", id.CertificateId.Name, id.CertificateId.ResourceGroup)
+		}
+		return fmt.Errorf("failed reading App Service Certificate %q (Resource Group %q): %+v", id.CertificateId.Name, id.CertificateId.ResourceGroup, err)
+	}
+
+	if certDetails.Thumbprint == nil {
+		return fmt.Errorf("could not read thumbprint from certificate %q (resource group %q): %+v", id.CertificateId.Name, id.CertificateId.ResourceGroup, err)
+	}
+	thumbprint := certDetails.Thumbprint
+
+	binding, err := client.GetHostNameBinding(ctx, id.HostnameBindingId.ResourceGroup, id.HostnameBindingId.SiteName, id.HostnameBindingId.Name)
+	if err != nil {
+		if utils.ResponseWasNotFound(binding.Response) {
+			return fmt.Errorf("retrieving Custom Hostname Binding %q (App Service %q / Resource Group %q): %+v", id.HostnameBindingId.Name, id.HostnameBindingId.SiteName, id.HostnameBindingId.ResourceGroup, err)
+		}
+		return fmt.Errorf("retrieving Custom Hostname Certificate Binding %q with certificate name %q (App Service %q / Resource Group %q): %+v", id.HostnameBindingId.Name, id.HostnameBindingId.SiteName, id.HostnameBindingId.SiteName, id.HostnameBindingId.ResourceGroup, err)
```

Comment:
`id.HostnameBindingId.SiteName` is here twice?

```suggestion
		return fmt.Errorf("retrieving Custom Hostname Certificate Binding %q with certificate name %q (App Service %q / Resource Group %q): %+v", id.HostnameBindingId.Name, id.CertificateId.Name, id.HostnameBindingId.SiteName, id.HostnameBindingId.ResourceGroup, err)
```